### ISSUE TITLE
Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows

### ISSUE THREAD

**Issue by wenfengyu** [tensorflow/tensorflow#46038]
After installing tensorflow2.4rc4 version on windows10, cudnn initialization error appears when running the training code. 
error info: Unimplemented: kernel reported driver version not implemented on Windows

tensorflow version:  tensorflow 2.4.0rc4
os :  windows 10
cuda version: 11.0
python version: 3.8
GPU: RTX3070 

2020-12-29 18:23:45.222526: E tensorflow/stream_executor/cuda/cuda_dnn.cc:336] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2020-12-29 18:23:45.222972: E tensorflow/stream_executor/cuda/cuda_dnn.cc:340] Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows
2020-12-29 18:23:45.225297: E tensorflow/stream_executor/cuda/cuda_dnn.cc:336] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2020-12-29 18:23:45.225401: E tensorflow/stream_executor/cuda/cuda_dnn.cc:340] Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows
2020-12-29 18:23:45.225508: F tensorflow/core/kernels/conv_grad_input_ops.cc:1173] Check failed: stream->parent()->GetConvolveBackwardDataAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(stream->parent()), &algorithms)

**Comment by wenfengyu** [comment]
cudnn version:  cudnn-11.0-windows-x64-v8.0.4.30

**Comment by Saduf2019** [comment]
@wenfengyu 
Please try setting a hard limit on the total GPU memory as shown in [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and check if it helps.
Pleasealso refer to similar issues: #45779, #45635, #42298.

**Comment by wenfengyu** [comment]
I tried to limit the hard limit on the total GPU memory many times, but still not fixed this issue, but use tf-nightly==2.5.0,  it's not this issue, but tf-nightly==2.5.0 can't use activation='tanh' in the Conv2D function. if use activation='tanh' in Conv2D, then the tranning process will crash. BTW, I try to use CUDA11.0 + cudnn-10.2-windows10-x64-v8.0.5.39, the tensorfow==2.4.0rc4 have not this issue(Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED...), but if use Conv2D, then will be no gradient drop during training.

**Comment by wenfengyu** [comment]
I use the follow set GPU memory code let it can not show this issue. (use  tensorflow2.4.0rc4 + CUDA11.0 + cudnn-11.0-windows-x64-v8.0.4.30)
config = tf.compat.v1.ConfigProto()
config.gpu_options.per_process_gpu_memory_fraction = 0.4
session = tf.compat.v1.Session(config=config)

but there will be no gradient drop during training when use Conv2D.

**Comment by wenfengyu** [comment]
use tf-nightly==2.5.0, gradient drop is normal during training when use Conv2D, but can't use activation='tanh' in the Conv2D function,  the traning process will be crash

**Comment by Tuxius** [comment]
I did also run into the issue:

`2021-01-03` 18:22:53.254748: E tensorflow/stream_executor/cuda/cuda_dnn.cc:336] Could not create cudnn handle: `CUDNN_STATUS_NOT_INITIALIZED`
`2021-01-03` 18:22:53.254813: E tensorflow/stream_executor/cuda/cuda_dnn.cc:340] Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows`

However, after rebooting Windows it worked fine, only with the second run after the reboot I got the error. I downgraded from TF 2.4.0 to 2.3.0, which resolved this issue for me.

**Comment by wenfengyu** [comment]
I also downgraded from TF 2.4.0 to 2.3.0, not this issue, but will be no gradient drop during training when use Conv2D.
